src/test/java/org/fogbowcloud/manager/core/AaControllerTest.java

```diff
@@ -48,112 +48,174 @@ public void setUp() {
     	
     }
     
+    //test case: Check if authenticate method throws no exception when the federation token is valid.
     @Test
     public void testAuthenticate() throws UnauthenticatedUserException {
+    	//set up
     	Mockito.when(this.federationIdentityPluginMock.isValid(Mockito.anyString())).thenReturn(true);
+    	
+    	//exercise/verify
     	this.aaController.authenticate(Mockito.anyString());
     }
     
+    //test case: Check if authenticate method throws Unauthenticated exception when the federation token is valid. 
     @Test (expected = UnauthenticatedUserException.class)
     public void testAuthenticateWhenUnauthenticatedUserException() throws UnauthenticatedUserException {
+    	//set up
     	Mockito.when(this.federationIdentityPluginMock.isValid(Mockito.anyString())).thenReturn(false);
+    	
+    	//exercise/verify
     	this.aaController.authenticate(Mockito.anyString());
     }
     
+    //test case: Check if authenticate method throws no exception when the operation is valid.
     @Test
-    public void testAuthorizeOnInstance() throws UnauthorizedRequestException {
+    public void testAuthorizeOnInstanceType() throws UnauthorizedRequestException {
+    	//set up
     	Mockito.when(this.authorizationPluginMock.isAuthorized(
     			Mockito.any(), 
     			Mockito.any(), 
     			Mockito.any(InstanceType.class))).thenReturn(true);
+    	
+    	//exercise/verify
     	this.aaController.authorize(Mockito.any(), Mockito.any(), Mockito.any(InstanceType.class));
     }
     
+    //test case: Check if authenticate method throws no exception when the operation is valid.
     @Test
-    public void testAuthorizeOnOrder() throws FogbowManagerException {
+    public void testAuthorizeOnOrderType() throws FogbowManagerException {
+    	//set up
     	Mockito.when(this.authorizationPluginMock.isAuthorized(
     			Mockito.any(), 
     			Mockito.any(), 
     			Mockito.any(Order.class))).thenReturn(true);
+    	
+    	//exercise/verify
     	this.aaController.authorize(Mockito.any(), Mockito.any(), Mockito.any(Order.class));
     }
     
+    //test case: Check if authenticate method throws no exception when the operation is valid.
     @Test
     public void testAuthorize() throws FogbowManagerException {
+    	//set up
     	Mockito.when(this.authorizationPluginMock.isAuthorized(
     			Mockito.any(), 
     			Mockito.any())).thenReturn(true);
+    	
+    	//exercise/verify
     	this.aaController.authorize(Mockito.any(), Mockito.any());
     }
     
+    //test case: Check if authenticate method throws Unauthenticated exception when the federation token is valid. 
```

Comment:
"... token is valid" or "token is invalid"? check the order test cases.

Reply:
copy-paste trouble